lib/cuda_color_spinor_field.cpp

```diff
@@ -574,6 +574,21 @@ namespace quda {
     backed_up = false;
   }
 
+  void cudaColorSpinorField::prefetch(QudaFieldLocation mem_space, cudaStream_t stream) const
+  {
+
+    // conditionals based on destructor
+    if (is_prefetch_enabled() && alloc && mem_type == QUDA_MEMORY_DEVICE) {
+      int dev_id = 0;
+      if (mem_space == QUDA_CUDA_FIELD_LOCATION) dev_id = comm_gpuid();
+      else if (mem_space == QUDA_CPU_FIELD_LOCATION) dev_id = cudaCpuDeviceId;
+      else errorQuda("Invalid QudaFieldLocation.");
+
+      cudaMemPrefetchAsync(v, bytes, dev_id, stream);
+      if (precision == QUDA_HALF_PRECISION || precision == QUDA_QUARTER_PRECISION) cudaMemPrefetchAsync(norm, norm_bytes, dev_id, stream);
```

Comment:
Can you change this to `(precision == QUDA_HALF_PRECISION || precision == QUDA_QUARTER_PRECISION) && norm_bytes > 0`.  This will allow for future optimization when doing fixed-point fields without a norm field (e.g., V fields).